### DISCUSSION TITLE
posrgresql: varchar length and column discription is lost in prisma generate output

### DISCUSSION THREAD

**Discussion by xiangnanscu** [prisma/prisma#12290] [Q&A] [answered]
could we have an option to preserve them?

**Comment by janpio** [comment]
What do you mean by "prisma generate output"? Can you provide an example what is happening, and what you would want to see instead?

**Reply by xiangnanscu** [reply]
say I create a table `ttt`:
```
CREATE TABLE "ttt" (
    "id" SERIAL unique,
    "name" varchar(33) 
);
comment on column ttt.name is '姓名';
```
and then in psql I run `\d+ ttt`, will get:
```
 Column |         Type          | Collation | Nullable |             Default             | Storage  | Stats target | Description 
--------+-----------------------+-----------+----------+---------------------------------+----------+--------------+-------------
 id     | integer               |           | not null | nextval('ttt_id_seq'::regclass) | plain    |              | 
 name   | character varying(33) |           |          |                                 | extended |              | 姓名
Indexes:
    "ttt_id_key" UNIQUE CONSTRAINT, btree (id)
```
then I use `prisma db pull`, will get `schema.prisma` like:
```
generator client {
  provider = "prisma-client-js"
}

datasource db {
  provider = "postgresql"
  url      = env("DATABASE_URL")
}

model ttt {
  id   Int     @unique @default(autoincrement())
  name String? @db.VarChar(33)
}
```
now,  as you can see, the `Description` info `姓名` is lost in `schema.prisma`. Next, in prisma github source code `prisma/packages/generator-helper/src/dmmf.ts`, the `Field` is defined as:
```
  export interface Field {
    kind: FieldKind
    name: string
    isRequired: boolean
    isList: boolean
    isUnique: boolean
    isId: boolean
    isReadOnly: boolean
    isGenerated?: boolean // does not exist on 'type' but does on 'model'
    isUpdatedAt?: boolean // does not exist on 'type' but does on 'model'
    /**
     * Describes the data type in the same the way is is defined in the Prisma schema:
     * BigInt, Boolean, Bytes, DateTime, Decimal, Float, Int, JSON, String, $ModelName
     */
    type: string
    dbNames?: string[] | null
    hasDefaultValue: boolean
    default?: FieldDefault | string | boolean | number
    relationFromFields?: string[]
    relationToFields?: any[]
    relationOnDelete?: string
    relationName?: string
    documentation?: string
    [key: string]: any // safe net for additional new props
  }
```
as you can see, no `maxlength` and no `Description` defined for postgresql. Thus there's no way for community generators like `prisma-dbml-generator` to get these infomation.

**Reply by janpio** [reply, marked as answer]
Ok got it, then we have two different issues here:

1. Prisma does not handle any comments or descriptions on columns yet, so this is not visible in the Prisma schema at all unfortunately. There are multiple issues about this as a feature request, searching for "comment" should surface some, one example: https://github.com/prisma/prisma/issues/8703
2. In the Prisma schema the lenth information is persisted as `@db.VarChar(33)`, a "native type". You are correct that this is not made available in the generated type or in the DMMF (the schema representation that is handed to community generators). The second bit about DMMF exists as an issue as well https://github.com/prisma/prisma/issues/10252 (which even has a PR we are discussing right now).

I don't think we have an issue about surfacing both information (native types information and column description/comment) in the generated Prisma Client types - that would be a good issue to create.

**Reply by janpio** [reply]
We actually have an issue about putting comments (from PSL) into the generated Prisma Client types: https://github.com/prisma/prisma/issues/5113

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
